### ISSUE TITLE
Unable to retrieve new Expression set version using CLI

### ISSUE THREAD

**Issue by svpsiva** [forcedotcom/cli#2823]
> **Note** 
> Before you submit your issue, make sure that:
> - You're using the latest version of Salesforce CLI.
> - You've searched both open and closed issues for related posts.
> - You've used the `doctor` command to diagnose common issues. 
> - You understand that GitHub Issues don't adhere to any agreement or SLA.
>   - If you require immediate assistance, use official channels such as Salesforce Customer Support.

### Steps To Reproduce
1. Create a new Expression set in Scratch org, activate it
2. Retrieve the ExpressionSetVersionDefinition to local project
3. Create a new version of Expression set in the scratch org
4. Preview the changes
sf project retrieve preview -o MyScratchOrg

No conflicts found.
No files will be deleted.
Will Retrieve [3] files.
 Type Fullname  Path 
 ─────────────────────────────── ────────────────────── ──── 
 DecisionMatrixDefinitionVersion Fixed_deposit_rates_V1 
 ExpressionSetDefinitionVersion FD_calculator_V1 
 ExpressionSetDefinitionVersion FD_calculator_V1CLONE0 

5. Try to retrieve the latest changes: 
sf project retrieve start -o MyScratchOrg

Retrieving v59.0 metadata from [test-sc0yaqj7y2ml@example.com](mailto:test-sc0yaqj7y2ml@example.com) using the v60.0 SOAP API
Preparing retrieve request... Succeeded
 ›  Warning: Nothing retrieved

### Expected result
SFDX cli should have retrieved the new version of expression set. 

### Actual result
Nothing was retrieved

**Comment by svpsiva** [comment]
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.36.8",
  "nodeVersion": "node-v20.11.1",
  "osVersion": "Darwin 23.4.0",
  "rootPath": "/Users/ssaripalli/.local/share/sf/client/2.36.8-fa01960",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.13 (core)",
    "@oclif/plugin-commands 3.2.2 (core)",
    "@oclif/plugin-help 6.0.20 (core)",
    "@oclif/plugin-not-found 3.1.1 (core)",
    "@oclif/plugin-plugins 5.0.5 (core)",
    "@oclif/plugin-search 1.0.20 (core)",
    "@oclif/plugin-update 4.2.3 (core)",
    "@oclif/plugin-version 2.0.16 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.15 (core)",
    "@oclif/plugin-which 3.1.7 (core)",
    "@salesforce/cli 2.36.8 (core)",
    "apex 3.1.3 (core)",
    "auth 3.5.1 (core)",
    "data 3.2.4 (core)",
    "deploy-retrieve 3.5.3 (core)",
    "info 3.1.2 (core)",
    "limits 3.2.1 (core)",
    "marketplace 1.1.0 (core)",
    "org 3.6.3 (core)",
    "packaging 2.3.0 (core)",
    "schema 3.2.0 (core)",
    "settings 2.1.2 (core)",
    "signups 2.1.2 (user)",
    "sobject 1.2.1 (core)",
    "source 3.2.2 (core)",
    "telemetry 3.1.17 (core)",
    "templates 56.1.0 (core)",
    "trust 3.5.1 (core)",
    "user 3.4.3 (core)"
  ]
}

**Comment by WillieRuemmele** [comment]
I was able to reproduce this
with project-scratch-def.json of 
```json
{
    "orgName": "Sample Org",
    "edition": "developer",
    "features": [
        "ADVANCEDDECISIONTABLE"
    ]
}
```

app launcher -> expression sets -> create -> save -> activate

```bash
 ➜  sf project retrieve start                 
Retrieving v60.0 metadata from test-vxlzforpe1qr@example.com using the v60.0 SOAP API
Preparing retrieve request... Succeeded

Retrieved Source
======================================================================================================================================================
| State   Name                            Type                    Path                                                                                 
| ─────── ─────────────────────────────── ─────────────────────── ──────────────────────────────────────────────────────────────────────────────────── 
| Created test                            ExpressionSetDefinition force-app/main/default/expressionSetDefinition/test.expressionSetDefinition-meta.xml 
| Created ExpressionSetConsole_UtilityBar FlexiPage               force-app/main/default/flexipages/ExpressionSetConsole_UtilityBar.flexipage-meta.xml 
---
Increment the version by 1 in the org
---
➜  dreamhouse-lwc git:(main) ✗  hub:(DevHub) scratch:(test-vxlzforpe1qr@example.com)
 ➜  sf project retrieve preview                   

No conflicts found.

No files will be deleted.

Will Retrieve [4] files.
 Type                           Fullname                  Path 
 ────────────────────────────── ───────────────────────── ──── 
 ExpressionSetDefinitionVersion test_V1                        
 ExpressionSetDefinitionVersion test_V1CLONE0                  
 ExpressionSetDefinitionVersion test_V1CLONE0CLONE0            
 ExpressionSetDefinitionVersion test_V1CLONE0CLONE0CLONE0      

No files were ignored. Update your .forceignore file if you want to ignore certain files.
➜  dreamhouse-lwc git:(main) ✗  hub:(DevHub) scratch:(test-vxlzforpe1qr@example.com)
 ➜  sf project retrieve start                 
Retrieving v60.0 metadata from test-vxlzforpe1qr@example.com using the v60.0 SOAP API
Preparing retrieve request... Succeeded
 ›   Warning: Nothing retrieved
```

**Comment by twardegaSFDC** [comment]
This issue applies to ExpressionSetDefinitionVersion and DecisionMatrixDefinitionVersion, both for retrieve and deploy using Salesforce CLI - tested with the latest version 2.38.7. The following example manifest file does not retrieve from org, and does not deploy to org:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Package xmlns="http://soap.sforce.com/2006/04/metadata">
    <types>
        <members>DAWSON_Case_Type_Mapping_V5</members>
        <members>DAWSON_Matter_Queue_Assignment_V1</members>
        <members>DAWSON_Event_Code_Mapping_V3</members>
        <members>DAWSON_Trial_Location_City_Mapping_V2</members>
        <name>DecisionMatrixDefinitionVersion</name>
    </types>
    <types>
        <members>OCC_Dawson_Contact_ES_V1</members>
        <members>DAWSONApplicableStagingMatter_V1CLONE1</members>
        <members>OCC_Dawson_Matter_ES_V1CLONE0</members>
        <members>OCC_Dawson_Legal_File_ES_V1</members>
        <members>OCC_Dawson_Practitioner_ES_V1</members>
        <name>ExpressionSetDefinitionVersion</name>
    </types>
	<version>60.0</version>
</Package>
```

**Comment by pgumiela-wbd** [comment]
@mshanemc, @shetzel, @WillieRuemmele, before we close this issue, can we ensure it also works for the sub-expression sets? I was testing the latest release candidate to retrieve `expressionSetDefinition` records, but I couldn't deploy them back (although `decisionMatrixDefinition` records were included) because of the following error:

`Couldn't find dependent components : {SUB_EXPRESSION=[9QA7Z000000Kz8VWAS, 9QA7Z000000Kz8WWAS, 9QA7Z000000Kz8XWAS]}`

I've raised support ticket #467793105 because it fails both when trying to deploy it directly and when using Package Manager. There's still much to be done to support these metadata types properly because even the legacy vlocity-build managed this (and it also deployed matrix row data, not just headers).

CC: @svpsiva, @twardegaSFDC

**Comment by jshackell-sfdc** [comment]
This issue is fixed in version 2.49.7 (July 10, 2024).

**Comment by pgumiela-wbd** [comment]
@jshackell-sfdc, before we close this issue, can you please check my comment about sub-expression sets? Has this scenario been tested?
